libs/community/langchain_community/vectorstores/aperturedb.py

```diff
@@ -0,0 +1,516 @@
+    ) -> List[Document]:
+        """Returns similar documents to the vector that also have diversity
+
+        This algorithm balances relevance and diversity in the search results.
+
+        Args:
+            embedding: Embedding vector to search for.
+            k: Number of results to return.
+            fetch_k: Number of results to fetch.
+            lambda_mult: Lambda multiplier for MMR.
+
+        Returns:
+            List of Document objects ordered by decreasing similarity/diversty.
+        """
+        from aperturedb.Descriptors import Descriptors
+
+        descriptors = Descriptors(self.connection)
+        start_time = time.time()
+        descriptors.find_similar_mmr(
+            set=self.descriptor_set,
+            vector=embedding,
+            k_neighbors=k,
+            fetch_k=fetch_k,
+            lambda_mult=lambda_mult,
+        )
+        self.logger.info(
+            f"ApertureDB similarity search mmr took {time.time() - start_time} seconds"
+        )
+        return [self._descriptor_to_document(d) for d in descriptors]
+
+    @classmethod
+    @override
+    def from_texts(
+        cls: Type[ApertureDB],
+        texts: List[str],
+        embedding: Embeddings,
+        metadatas: Optional[List[dict]] = None,
+        **kwargs: Any,
+    ) -> ApertureDB:
+        """Creates a new vectorstore from a list of texts
+
+        Args:
+            texts: List of text strings
+            embedding: Embeddings object as for constructing the vectorstore
+            metadatas: Optional list of metadatas associated with the texts.
+            **kwargs: Additional arguments to pass to the constructor
+        """
+        store = cls(embeddings=embedding, **kwargs)
+        store.add_texts(texts, metadatas)
+        return store
+
+    @classmethod
+    @override
+    def from_documents(
+        cls: Type[ApertureDB],
+        documents: List[Document],
+        embedding: Embeddings,
+        **kwargs: Any,
+    ) -> ApertureDB:
+        """Creates a new vectorstore from a list of documents
+
+        Args:
+            documents: List of Document objects
+            embedding: Embeddings object as for constructing the vectorstore
+            metadatas: Optional list of metadatas associated with the texts.
+            **kwargs: Additional arguments to pass to the constructor
+        """
+        store = cls(embeddings=embedding, **kwargs)
+        store.add_documents(documents)
+        return store
+
+    @classmethod
+    def delete_vectorstore(class_, descriptor_set: str) -> None:
+        """Deletes a vectorstore and all its data from the database
+
+        Args:
+            descriptor_set: The name of the descriptor set to delete
+        """
+        from aperturedb.Utils import Utils, create_connector
+
+        db = create_connector()
+        utils = Utils(db)
+        utils.remove_descriptorset(descriptor_set)
+
+    @classmethod
+    def list_vectorstores(class_) -> None:
```

Comment:
We're thinking about adding an abstraction for the control plane that will allow listing, creating, deleting indices. Let me know if you would be interested in testing it out

Reply:
Yes, please let me know.